src/ngraph_builder.cc

```diff
@@ -41,6 +41,36 @@ const static std::map<tf::DataType, ngraph::element::Type> TF_NGRAPH_TYPE_MAP =
      {tf::DataType::DT_UINT16, ng::element::u16},
      {tf::DataType::DT_BOOL, ng::element::boolean}};
 
+// Helper for storing ops in ng_op_map
+static void SaveNgOp(Builder::OpMap& map, std::string key, 
+                     shared_ptr<ng::Node> value, 
+                     size_t index = 0) { 
+  if (map[key].size() <= index) {
+    map[key].resize(index + 1);
+  }
+  map[key][index] = value; 
+}
+
+// Helper for lookup in ng_op_map 
+static shared_ptr<ng::Node> GetNgOp(Builder::OpMap &map, 
+                                    std::string key, 
+                                    size_t index = 0) {
+  return map.at(key).at(index);
+}
+
+#define TRY_FETCH_NG_OP(dest, index) ({ \
```

Comment:
Since you are calling a function (which should have the exception handling) - no point using this MACRO. Rather have the function return tf::Status object and return the value as one of the last argument.